alternator/streams.cc

```diff
@@ -270,7 +270,17 @@ struct sequence_number {
 
         using namespace boost::multiprecision;
 
-        uint128_t hi = uint64_t(num.uuid.get_most_significant_bits());
+        /**
+         * #7424 - aws sdk assumes sequence numbers are
+         * monotonically growing bigints. 
```

Comment:
timestamps in cdc$time are non-decreasing is that enough?

Reply:
Timestamps are monotonically growing, and timeuuids are sorted as such, _but_ their bit representation if viewed just as 128 unsigned bytes are _not_, because the msb portion is bitswapped around (reflected in the comparator). What we do here is simply unswap the bits so we get the "actual" value <timestamp>|lsb as a 128 bit number.

Reply:
it is possible for two rows in CDC Log to have cdc$time that translates to the same timestamp. cdc$time is a timeuuid especially because of that. To resolve conflicts when two operations have the same timestamp. Thus, timestamps in cdc$time are non-decreasing but not strictly growing.

Reply:
Maybe the resulting 128-bit values after the transformation of msb will be sorted in the same way for two timeuuids that share the same timestamp but I don't know. That's why I'm asking. Plus the non-decreasing vs strictly growing thing.

Reply:
Yes, that is the effect. We just ensure the 128-bit int is the bigint representation of how the timeuuid was sorted (the effective bit comparison order).